File: leetcode/1496. Path Crossing.java
class Solution {
    public boolean isPathCrossing(String path) {
        // use x * 1001 + y change coordinate to num
        Set<Integer> record = new HashSet<>();
        record.add(0);

        int x = 0, y = 0;
        for (int i = 0; i < path.length(); ++i) {
            char c = path.charAt(i);
            if (c == 'N')
                ++y;
            else if (c == 'S')
                --y;
            else if (c == 'E')
                ++x;
            else if (c == 'W')
                --x;

            int cood = x * 1001 + y;

            if (record.contains(cood))
                return true;

            record.add(cood);
        }
        return false;
    }
}

File: leetcode/1496. Path Crossing.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        record = set()
        record.add("0,0")

        x, y = 0, 0
        for p in path:
            if p == 'N':
                y += 1
            elif p == 'S':
                y -= 1
            elif p == 'W':
                x -= 1
            elif p == 'E':
                x += 1

            cood = str(x) + ',' + str(y)
            print(cood)
            
            if cood in record:
                return True
            
            record.add(cood)
        
        return False